File: python/components/software.py
class Software:
    def __init__(self, version="", license_key="", size_mb=0.0):
        self._version = version
        self._license_key = license_key
        self._size_mb = size_mb

    # Setters
    def set_version(self, version):
        self._version = version

    def set_license_key(self, license_key):
        self._license_key = license_key

    def set_size_mb(self, size_mb):
        self._size_mb = size_mb

    # Getters
    def get_version(self):
        return self._version

    def get_license_key(self):
        return self._license_key

    def get_size_mb(self):
        return self._size_mb

File: java/components/Software.java
package components;

public abstract class Software extends Component {
    protected String version;
    protected String licenseKey;
    protected double sizeMB;

    public Software(String id, String manufacturer, String model, double price, String version, String licenseKey,
            double sizeMB) {
        super(id, manufacturer, model, price);
        this.version = version;
        this.licenseKey = licenseKey;
        this.sizeMB = sizeMB;
    }

    public String getVersion() {
        return version;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public String getLicenseKey() {
        return licenseKey;
    }

    public void setLicenseKey(String licenseKey) {
        this.licenseKey = licenseKey;
    }

    public double getSizeMB() {
        return sizeMB;
    }

    public void setSizeMB(double sizeMB) {
        this.sizeMB = sizeMB;
    }
}